dencoding/yaml_decoder_test.go

```diff
@@ -98,12 +98,16 @@ key2: value6
 	})
 
 	t.Run("YamlAliases", func(t *testing.T) {
-		b := []byte(`foo: &foo
+		b := []byte(`foo: &foofoo
   bar: 1
-  baz: "baz"
+  baz: &baz "baz"
```

Comment:
Instead of using a top-level anchor, I used one inside a map to make sure this case is also covered.